From 353efbedd40f27e4f6ae5a653ef5ee9c2e00829e Mon Sep 17 00:00:00 2001
From: Sebastian Fricke <sebastian.fricke@posteo.net>
Date: Sat, 6 Feb 2021 08:23:14 +0100
Subject: [PATCH 2/3] arm64: dts: Add the OV13850 and OV4689 to dts

Add the two offical friendlyElec cameras to the device tree.
Mostly a copy of what is found on the friendlyElec fork of the rockchip
downstream BSP kernel tree.
Additionally, increase the temperature limits on the CPU

Signed-off-by: Sebastian Fricke <sebastian.fricke@posteo.net>
---
 .../boot/dts/rockchip/rk3399-nanopc-t4.dts    | 165 ++++++++++++++++++
 arch/arm64/boot/dts/rockchip/rk3399.dtsi      |   7 +-
 2 files changed, 169 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts b/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts
index e0d75617bb7e..889d65f867d4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-nanopc-t4.dts
@@ -32,6 +32,54 @@ vcc5v0_host0: vcc5v0-host0 {
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	ov13850_avdd_2p8v: 2p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov13850_avdd";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		regulator-always-on;
+	};
+
+	ov13850_dovdd_1p8v: 1p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov13850_dovdd";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+	};
+
+	ov13850_dvdd_1p2v: 1p2v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov13850_dvdd";
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+		regulator-always-on;
+	};
+
+	ov4689_avdd_2p8v: 2p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov4689_avdd";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		regulator-always-on;
+	};
+
+	ov4689_dovdd_1p8v: 1p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov4689_dovdd";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+	};
+
+	ov4689_dvdd_1p2v: 1p2v {
+		compatible = "regulator-fixed";
+		regulator-name = "ov4689_dvdd";
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+		regulator-always-on;
+	};
+
 	adc-keys {
 		compatible = "adc-keys";
 		io-channels = <&saradc 1>;
@@ -99,6 +147,61 @@ &pcie0 {
 	vpcie3v3-supply = <&vcc3v3_sys>;
 };
 
+
+&i2c1 {
+	status = "okay";
+	ov13850p0: ov13850@10 {
+		compatible = "ovti,ov13850";
+		status = "okay";
+		reg = <0x10>;
+		clocks = <&cru 0x89>;
+		clock-names = "xvclk";
+		avdd-supply = <&ov13850_avdd_2p8v>;
+		dovdd-supply = <&ov13850_dovdd_1p8v>;
+		dvdd-supply = <&ov13850_dvdd_1p2v>;
+
+		reset-gpios = <&gpio2 27 0>;
+		pwdn-gpios = <&gpio2 28 0>;
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
+		pinctrl-0 = <&cam0_default_pins &cif_clkout_a>;
+		pinctrl-1 = <&cam0_default_pins>;
+
+		port {
+			ucam_out0a: endpoint {
+				remote-endpoint = <&mipi_in_ucam0a>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+};
+
+&i2c2 {
+	status = "okay";
+	ov4689p1: ov4689@36 {
+		compatible = "ovti,ov4689";
+		status = "okay";
+		reg = <0x36>;
+		clocks = <&cru 0x89>;
+		clock-names = "xvclk";
+		avdd-supply = <&ov4689_avdd_2p8v>;
+		dovdd-supply = <&ov4689_dovdd_1p8v>;
+		dvdd-supply = <&ov4689_dvdd_1p2v>;
+
+		reset-gpios = <&gpio0 8 0>;
+		pwdn-gpios = <&gpio0 12 0>;
+		pinctrl-names = "rockchip,camera_default", "rockchip,camera_sleep";
+		pinctrl-0 = <&cam1_default_pins &cif_clkout_b>;
+		pinctrl-1 = <&cam1_default_pins>;
+
+		port {
+			ucam_out1a: endpoint {
+				remote-endpoint = <&mipi_in_ucam1a>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+};
+
 &pinctrl {
 	ir {
 		ir_rx: ir-rx {
@@ -106,6 +209,68 @@ ir_rx: ir-rx {
 			rockchip,pins = <0 RK_PA6 1 &pcfg_pull_none>;
 		};
 	};
+
+	cam_pins {
+		cif_clkout_a: cif-clkout-a {
+			rockchip,pins = <2 11 3 &pcfg_pull_none>;
+		};
+
+		cif_clkout_b: cif_clkout_b {
+			rockchip,pins = <2 15 3 &pcfg_pull_none>;
+		};
+
+		cif_clkout_a_sleep: cif-clkout-a-sleep {
+			rockchip,pins = <2 11 0 &pcfg_pull_none>;
+		};
+
+		cam0_default_pins: cam0-default-pins {
+			rockchip,pins = <2 28 0 &pcfg_pull_down>, <2 27 0 &pcfg_pull_none>;
+		};
+
+		cam1_default_pins: cam1-default-pins {
+			rockchip,pins = <0 12 0 &pcfg_pull_down>, <0  8 0 &pcfg_pull_none>;
+		};
+	};
+};
+
+&mipi_dphy_rx0 {
+	status = "okay";
+};
+
+&mipi_dsi1 {
+	status = "okay";
+};
+
+&isp0_mmu {
+	status = "okay";
+};
+
+&isp0 {
+	status = "okay";
+
+	ports {
+		port@0 {
+			mipi_in_ucam0a: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&ucam_out0a>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+};
+
+&isp1 {
+	status = "okay";
+
+	ports {
+		port@0 {
+			mipi_in_ucam1a: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&ucam_out1a>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
 };
 
 &sdhci {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index f5dee5f447bb..58d2e6c5104e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -772,17 +772,17 @@ cpu_thermal: cpu {
 
 			trips {
 				cpu_alert0: cpu_alert0 {
-					temperature = <70000>;
+					temperature = <85000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
 				cpu_alert1: cpu_alert1 {
-					temperature = <75000>;
+					temperature = <95000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
 				cpu_crit: cpu_crit {
-					temperature = <95000>;
+					temperature = <100000>;
 					hysteresis = <2000>;
 					type = "critical";
 				};
@@ -1873,6 +1873,7 @@ mipi_dsi1: mipi@ff968000 {
 		rockchip,grf = <&grf>;
 		#address-cells = <1>;
 		#size-cells = <0>;
+		#phy-cells = <0>;
 		status = "disabled";
 
 		ports {
-- 
2.25.1

